packages/liveblocks-core/src/room.ts

```diff
@@ -456,25 +488,121 @@ type SubscribeFn<
   (type: "storage-status", listener: Callback<StorageStatus>): () => void;
 };
 
+export type GetThreadsOptions<TThreadMetadata extends BaseMetadata> = {
+  query?: {
+    metadata?: Partial<TThreadMetadata>;
+  };
+  since?: Date;
+};
+
+type CommentsApi<TThreadMetadata extends BaseMetadata = never> = {
+  /**
+   * @private
+   */
+  getThreads(options?: GetThreadsOptions<TThreadMetadata>): Promise<{
+    threads: ThreadData<TThreadMetadata>[];
+    inboxNotifications: InboxNotificationData[];
+    deletedThreads: ThreadDeleteInfo[];
+    deletedInboxNotifications: InboxNotificationDeleteInfo[];
+    meta: {
+      requestedAt: Date;
+    };
+  }>;
+
+  /**
+   * @private
+   */
+  getThread(options: { threadId: string }): Promise<
+    | {
+        thread: ThreadData<TThreadMetadata>;
+        inboxNotification?: InboxNotificationData;
+      }
+    | undefined
+  >;
+
+  /**
+   * @private
+   */
+  createThread(options: {
+    threadId: string;
+    commentId: string;
+    metadata: TThreadMetadata | undefined;
+    body: CommentBody;
+  }): Promise<ThreadData<TThreadMetadata>>;
+
+  /**
+   * @private
+   */
+  editThreadMetadata(options: {
+    metadata: PartialNullable<TThreadMetadata>;
+    threadId: string;
+  }): Promise<TThreadMetadata>;
+
+  /**
+   * @private
+   */
+  createComment(options: {
+    threadId: string;
+    commentId: string;
+    body: CommentBody;
+  }): Promise<CommentData>;
+
+  /**
+   * @private
+   */
+  editComment(options: {
+    threadId: string;
+    commentId: string;
+    body: CommentBody;
+  }): Promise<CommentData>;
+
+  /**
+   * @private
+   */
+  deleteComment(options: {
+    threadId: string;
+    commentId: string;
+  }): Promise<void>;
+
+  /**
+   * @private
+   */
+  addReaction(options: {
+    threadId: string;
+    commentId: string;
+    emoji: string;
+  }): Promise<CommentUserReaction>;
+
+  /**
+   * @private
+   */
+  removeReaction(options: {
+    threadId: string;
+    commentId: string;
+    emoji: string;
+  }): Promise<void>;
+};
+
 export type Room<
   TPresence extends JsonObject,
   TStorage extends LsonObject,
   TUserMeta extends BaseUserMeta,
   TRoomEvent extends Json,
 > = CommentsApi<any /* TODO: Remove this any by adding a proper thread metadata on the Room type */> & {
```

Comment:
Can be a fix in here for the `any` in `CommentsApi`? 

And/or, expose the `CommentsApi` type from the root folder to let users of the liveblocks sdk handle this themselves? Would help with extending that interface for custom hooks etc.